src/main/java/com/alrex/parcool/common/action/impl/Vault.java

```diff
@@ -162,7 +162,7 @@ public void onWorkingTickInLocalClient(PlayerEntity player, Parkourability parko
 		if (stepDirection == null) return;
 		player.setDeltaMovement(
 				stepDirection.x() / 10,
-				(stepHeight + 0.02) / this.getVaultAnimateTime(),
+				(stepHeight + (0.002 * (1.8 / player.getBbHeight()))) / (this.getVaultAnimateTime() * (player.getBbHeight() / 1.8)),
```

Comment:
Why do you multiply time by player size?
What do this change mean actually?

Reply:
ok

Reply:
ok i was trying to see whether i could remove the preemptive guessing at what other mods will do in this code, but i think it might be the only option as we need to give the player an appropriate y value based on how high the wall is. meaning we have to be the ones in control of the players motion value, and if any mod that messes with it, the vault will break. the only way i can think to address this issue is to preemptively change the value and guess what other mods are going to do, which is what we have now.

i can't think of any other way to improve all of these changes and will now stop working on the code (unless you can think of any better ways of doing things)